bevy_rapier3d/examples/ray_casting3.rs

```diff
@@ -90,9 +90,9 @@ pub fn cast_ray(
         let Ok(ray) = camera.viewport_to_world(camera_transform, cursor_position) else {
             return;
         };
-
+        let context = rapier_context.single();
         // Then cast the ray.
-        let hit = rapier_context.cast_ray(
+        let hit = context.cast_ray(
```

Comment:
We could avoid this indirection to `single()` if we implemented all methods of `RapierContext` on its systemParam ; but:
- it would require making the query each time we call a method
  - Probably not so much of an issue if queries are cached
- it would require to add another 400 lines of code, I already fear for the maintenance of current layer of indirection, I'd like to keep it at the minimum. a macro to generate those ended up a bit too complex ; maybe a procedural or a template step could be interesting.